react/src/pages/MaintenancePage.tsx

```diff
@@ -1,89 +1,203 @@
-import { useWebComponentInfo } from '../components/DefaultProviders';
-import Flex from '../components/Flex';
+import { SettingItemProps } from '../components/SettingItem';
+import SettingList from '../components/SettingList';
+import { useSuspendedBackendaiClient } from '../hooks';
+import { useSetBAINotification } from '../hooks/useBAINotification';
+import { usePainKiller } from '../hooks/usePainKiller';
 import { RedoOutlined } from '@ant-design/icons';
-import { Button, Card, Row, Col, Typography, theme } from 'antd';
-import React from 'react';
-import { useTranslation, Trans } from 'react-i18next';
+import { Button, Card } from 'antd';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { StringParam, useQueryParam, withDefault } from 'use-query-params';
 
-const MaintenancePage: React.FC = () => {
+type TabKey = 'maintenance';
+type SettingGroup = {
+  title: string;
+  settingItems: SettingItemProps[];
+};
+
+const tabParam = withDefault(StringParam, 'maintenance');
+
+const MaintenanceSettingList = () => {
+  // state
+  const [isRecalculating, setIsRecalculating] = useState(false);
+  const [isRescanning, setIsRescanning] = useState(false);
+  // hooks
   const { t } = useTranslation();
+  const { upsertNotification } = useSetBAINotification();
+  const baiClient = useSuspendedBackendaiClient();
+  const painKiller = usePainKiller();
+  // funcs
+  const recalculateUsage = async () => {
+    setIsRecalculating(true);
 
-  const { value, dispatchEvent } = useWebComponentInfo();
-  let parsedValue: {
-    recalculating: boolean;
-    scanning: boolean;
-  };
-  try {
-    parsedValue = JSON.parse(value || '');
-  } catch (error) {
-    parsedValue = {
-      recalculating: false,
-      scanning: false,
-    };
-  }
-  const { recalculating, scanning } = parsedValue;
+    const task = new Promise((resolve, reject) => {
+      baiClient.maintenance
+        .recalculate_usage()
+        .then(() => resolve(null))
+        .catch((err: any) => {
+          console.error(err);
+          if (err && err.message) {
+            upsertNotification({
+              message: err.message,
+              description: painKiller.relieve(err.title),
+              open: true,
+            });
+          }
+          reject(err);
+        })
+        .finally(() => setIsRecalculating(false));
+    });
 
```

Comment:
`baiClient.maintenance` already returns a promise, so you don't need to create a `new Promise`.